content/blog/2022/2022-09-01-schedule-cron-job-in-node.md

```diff
@@ -0,0 +1,319 @@
+
+We are using the method  `process.heapUsed` . The `heapUsed` refer to V8's memory use by our application. And `os.freemem()` shows available RAM, `os.totalmem()` show entire memory capacity.
+
+The log is saved in `.csv` format, with the date/time in the first column, memory usage in the second, and the memory available in the third. These data are recorded and saved in a generated `demo.csv` file at 15-second intervals.
+
+Run the script: `node index.js`
+
+Allow your application to run, you will notice a file named `demo.csv` is generated with content similar to the following:
+
+```csv
+2022-08-31T00:19:45.912Z, 8.495856, 10%
+2022-08-31T00:20:00.027Z, 7.083216, 10%
+2022-08-31T00:20:15.133Z, 7.139864, 9%
+2022-08-31T00:20:30.219Z, 7.188568, 12%
+2022-08-31T00:20:45.414Z, 7.23724, 11%
+```
+
+### 4. Deleting / Refreshing a Log File
+Consider a scenario where we are working with a large application that records the status of all activity in the log file. The log status file would eventually grow large and out of date. You can routinely delete the log file from the server. For instance, we could routinely delete the log status file using a job scheduler on the 25th of every month.
+
+In this example, you will be deleting the log status file that was previously created:
+```javascript
+const express = require("express");
+const cron = require("node-cron");
+const fs = require("fs");
+
+app = express();
+
+// remove the demo.csv file every twenty-first day of the month.
+cron.schedule("0 0 25 * *", function () {
+  console.log("---------------------");
+  console.log("deleting logged status");
+  fs.unlink("./demo.csv", err => {
+    if (err) throw err;
+    console.log("deleted successfully");
+  });
+});
+
+app.listen(3000, () => {
+  console.log("Application Listening.....");
+});
+```
+
+Notice the pattern used: `0 0 25 * *`.
+
+- minutes and hours as `0` and `0` (“00:00” - the start of the day).
+- specific day of the month as `25`.
+- month or day of the week isn't defined.
+
+Now, run the script: `node index.js`
+
+On the 25th of every month, your log status will be deleted with:
+
+`output:`
+```
+Application Listening.....
+---------------------
+Running Cron Job
+Error file successfully deleted
+```
+Switch cron expression to a shorter time interval - like every minute. To verify the task is been executed.
+
+Checking the application directory. You will notice the logstatus.txt file has been deleted.
+
+## Conclusion
+This article demonstrates how to schedule tasks on the Node.js server using node-cron, and the concept of using node-cron jobs to automate and schedule repetitive or future tasks. You can use this idea in both current and upcoming projects.
+
+There are other job scheduler tools accessible to node.js applications such as node-schedule, Agenda, Bree, Cron, and Bull. Be sure to assess each one to find the best fit for your specific project.
```

Comment:
In the conclusion, please link to the GitHub repo again so that readers know where they can find the code. It should link to where the code will be after your PR to the `code-examples` repo has been merged.